client/components/reviews-rating-stars/reviews-ratings-stars.tsx

```diff
@@ -0,0 +1,145 @@
+	const onSaveRating = ( _: unknown, index: number ) => {
+		setRatingState( index );
+		if ( onSelectRating ) {
+			onSelectRating( index );
+		}
+	};
+
+	useEffect( () => {
+		setRatingState( rating );
+		setHoverRating( rating );
+	}, [ rating ] );
+
+	const classNames = [ 'reviews-ratings-stars__star-bar' ];
+
+	switch ( size ) {
+		case 'medium-small':
+			classNames.push( 'reviews-ratings-stars__star-bar--medium-small' );
+			break;
+		case 'medium-large':
+			classNames.push( 'reviews-ratings-stars__star-bar--medium-large' );
+			break;
+	}
+
+	// Show numberOfReviews if the prop was passed
+	let numberOfReviewsText;
+	if ( numberOfReviews ) {
+		numberOfReviewsText = translate( '%(numberOfReviews) review', '%(numberOfReviews) reviews', {
+			count: numberOfReviews,
+			args: { numberOfReviews },
+		} );
+	}
+
+	function getStarsNumberSign( numberOfStars: number ) {
+		return translate( '%(numberToAdd)d star', '%(numberToAdd)d stars', {
+			count: numberOfStars,
+			args: { numberToAdd: numberOfStars },
+		} );
+	}
+
+	// Show currently selected rating if we're not showing rating count
+	let selectedRatingText;
+	if ( ratingState && ! numberOfReviews && showSelectedRating ) {
+		selectedRatingText = getStarsNumberSign( ratingState );
+	}
+
+	const ratingNormalized = ( rating * 100 ) / 20;
+
+	return (
+		<div
+			className={ classNames.join( ' ' ) }
+			onMouseEnter={ onMouseEnter }
+			onMouseLeave={ onMouseLeave }
+		>
+			{ simpleView ? (
+				<Rating rating={ ratingNormalized } />
+			) : (
+				<>
+					{ averageRating && (
+						<div className="reviews-ratings-stars__average">
+							<span className="screen-reader-text">{ translate( 'Average rating ' ) }</span>
+							{ averageRating }
+						</div>
+					) }
+					<div className="reviews-ratings-stars__stars" aria-hidden="false">
+						{ Array( MAX_RATING )
+							.fill( null )
+							.map( ( _, index ) => {
+								const value = index + 1;
+								const ratingStarsScreenReaderText = translate(
+									'Rate product %(value) star',
+									'Rate product %(value) stars',
+									{ count: value, args: { value } }
+								);
+
+								const starProps: StarProps & React.RefAttributes< SVGSVGElement > = {
+									rating: ratingState,
+									index: value,
+									hoverRating,
+									onMouseEnter: onStarMouseEnter,
+									onMouseLeave: onStarMouseLeave,
+									onClick: onSaveRating,
+									isInteractive: true,
+									tabIndex: 0,
+									ariaLabel: ratingStarsScreenReaderText as string,
+								};
+
+								return <Star { ...starProps } key={ value } />;
+							} ) }
+					</div>
+					{ numberOfReviewsText && (
+						<div className="reviews-ratings-stars__rating-count">{ numberOfReviewsText }</div>
+					) }
+					{ selectedRatingText && (
+						<div className="reviews-ratings-stars__selected-rating">{ selectedRatingText }</div>
+					) }
+				</>
+			) }
+			{ children && children }
```

Comment:
Any reason to not put just `{children}`?

Reply:
Not really, it comes from the original implementation and I think it is an old pattern to check if children exists, but I don't think is required and it hurts readability so I have removed it in d0818bd3256e92faa2659619e1a95a469858157c